File: GFG/Greedy/activitysel.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<utility>
using namespace std;

bool psort(pair<int,int> a,pair<int,int> b){
    return (a.second<b.second);
}
int activity(vector<pair<int,int>> &arr){
    sort(arr.begin(),arr.end(),psort);
    int prev=0;
    int res=1;
    for(int curr=1;curr<arr.size();curr++){
        if(arr[curr].first>=arr[prev].second){
            prev=curr;
            res++;
        }
    }
    return res;
}

int main(){
    vector<pair<int,int>> arr {{12,19},{10,20},{20,30},{7,11}};
    cout<<activity(arr);
    return 0;
}

File: GFG/Greedy/activitysel.py
def psort(tup1,tup2):
    return tup1[1]<tup2[1]

def activity(arr):
    arr=sorted(arr,key=lambda x:x[1])
    prev=0
    res=1
    for i in range(len(arr)):
        if(arr[i][0]>=arr[prev][1]):
            res+=1
            prev=i
    return res
    
    
arr=[(12,19),(10,20),(20,30),(7,11)]
print(activity(arr))
    